src/error.rs

```diff
@@ -24,6 +27,10 @@ pub enum EvalError<'tcx> {
     Unimplemented(String),
     DerefFunctionPointer,
     ExecuteMemory,
+    ArrayIndexOutOfBounds(Span, u64, u64),
+    Math(Span, ConstMathErr),
+    InvalidBitShiftRhs(PrimVal),
+    Overflow(PrimVal, PrimVal, mir::BinOp, PrimVal),
```

Comment:
Does this get ignored outside of `intrinsic_overflowing`? Again, the semantics are 2's complement + wrapping, unless explicit checking is done and reported with `Assert`.